tests/multi_syscall_test.rs

```diff
@@ -0,0 +1,288 @@
+use num_traits::{Num, Zero};
+use starknet_in_rust::utils::calculate_sn_keccak;
+use starknet_in_rust::EntryPointType;
+use starknet_in_rust::{
+    definitions::{block_context::BlockContext, constants::TRANSACTION_VERSION},
+    execution::{
+        execution_entry_point::ExecutionEntryPoint, CallInfo, CallType, OrderedEvent,
+        TransactionExecutionContext,
+    },
+    state::cached_state::CachedState,
+    state::{in_memory_state_reader::InMemoryStateReader, ExecutionResourcesManager},
+    utils::{Address, ClassHash},
+};
+use std::{collections::HashMap, sync::Arc, vec};
+
+#[test]
+fn test_multiple_syscall() {
+    //  Create program and entry point types for contract class
+    let program_data = include_bytes!("../starknet_programs/cairo1/multi_syscall_test.casm");
+    let contract_class: CasmContractClass = serde_json::from_slice(program_data).unwrap();
+
+    // Create state reader with class hash data
+    let mut contract_class_cache: HashMap<[u8; 32], _> = HashMap::new();
+
+    let address = Address(1111.into());
+    let class_hash: ClassHash = [1; 32];
+    let nonce = Felt252::zero();
+
+    contract_class_cache.insert(class_hash, contract_class.clone());
+    let mut state_reader = InMemoryStateReader::default();
+    state_reader
+        .address_to_class_hash_mut()
+        .insert(address.clone(), class_hash);
+    state_reader
+        .address_to_nonce_mut()
+        .insert(address.clone(), nonce);
+
+    // Create state from the state_reader and contract cache.
+    let mut state = CachedState::new(
+        Arc::new(state_reader),
+        None,
+        Some(contract_class_cache.clone()),
+    );
+
+    // Create an execution entry point
+    let calldata = [].to_vec();
+    let caller_address = Address(0000.into());
+    let entry_point_type = EntryPointType::External;
+
+    // Block for get_caller_address.
+    {
+        let call_info = test_syscall(
+            "caller_address",
+            address.clone(),
+            vec![],
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.retdata, vec![caller_address.clone().0])
+    }
+
+    // Block for get_contact_address.
+    {
+        let call_info = test_syscall(
+            "contract_address",
+            address.clone(),
+            vec![],
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![])
+    }
+    // Block for get_execution_info_syscall.
+    {
+        let call_info = test_syscall(
+            "execution_info_syscall",
+            address.clone(),
+            calldata.clone(),
+            caller_address.clone(),
+            entry_point_type,
+            class_hash,
+            &mut state,
+        );
+        assert_eq!(call_info.events, vec![]);
+    }
+
+    // Block for replace_class_syscall
```

Comment:
This is a tricky syscall because it replaces the class of the contract. This means the ABI will be different. You can delete this test because it may mess up the other tests